src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -650,28 +651,35 @@ public IEnumerable<string> Audiences
             {
                 if (_audiences == null)
                 {
-                    _audiences = new List<string>();
-#if NET45
-                    if (Payload.TryGetValue(JwtRegisteredClaimNames.Aud, out JToken value))
-                    {
-                        if (value.Type is JTokenType.String)
-                            _audiences = new List<string> { value.ToObject<string>() };
-                        else if (value.Type is JTokenType.Array)
-                            _audiences = value.ToObject<List<string>>();
-                    }
-#else
-                    if (Payload.TryGetValue(JwtRegisteredClaimNames.Aud, out JsonElement audiences))
+                    lock (_audiencesLock)
                     {
-                        if (audiences.ValueKind == JsonValueKind.String)
-                            _audiences = new List<string> { audiences.GetString() };
-
-                        if (audiences.ValueKind == JsonValueKind.Array)
+                        if (_audiences == null)
```

Comment:
do you need to check again for null? see line 652

Reply:
see other comment, this is a standard double checked lock pattern

Reply:
Yes it is necessary, only new up the list if its notnull.